src/message-manager.mts

```diff
@@ -0,0 +1,155 @@
+// Main function that is responsible for removing / adding countdowns
+// HOW COUNTDOWNS WORK:
+// It's very shrimple, we serialize/deserialize a json dictionary containing the "Countdowns" struct
+// Events are stored in a dictionary, with the key being the event name and value containing any other data about it
+
+import fs from "node:fs";
+import type { Client } from "discord.js";
+import { TextChannel, Message, EmbedBuilder } from "discord.js";
+import { ScheduledTask } from "node-cron";
+
+export type CountdownMessageInput = {
+    eventDate: Date;
+    eventName: string;
+    eventLink: string | null;
+};
+
+type Countdowns = {
+    // Channel of countdown
+    [channelId: string]: {
+        messageId: string;
+        events: {
+            // Events of countdown
+            [eventName: string]: {
+                eventDate: Date; // Date of countdown
+                eventLink: string; // Any link if given
+            };
+        };
+    };
+};
+
+// Deserialize messages
+const messageDictionary: Countdowns = fs.existsSync("./messages.json")
```

Comment:
Every time this is imported, you're going to read that file synchronously over and over + style guide

Reply:
Singleton?